src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/submission/BaseSubmissionController.java

```diff
@@ -202,24 +205,27 @@ public DTO update(
     public JsonResult<Boolean> uploadSubmissionResults(
             Principal principal,
             @RequestParam("submissionId") Long id,
+            @RequestParam(value = "archive", defaultValue = "false") boolean archive,
             @Valid UploadFileDTO uploadFileDTO
-    ) throws IOException, NotExistException, ValidationException {
+    ) throws IOException, NotExistException {
 
         LOGGER.info("uploading result files for submission with id='{}'", id);
-
-        JsonResult.ErrorCode errorCode;
-        Boolean hasResult;
         if (uploadFileDTO.getFile() == null) {
-            errorCode = JsonResult.ErrorCode.VALIDATION_ERROR;
-            hasResult = false;
-        } else {
-            submissionService.uploadResultsByDataOwner(id, uploadFileDTO.getLabel(), uploadFileDTO.getFile());
-            errorCode = JsonResult.ErrorCode.NO_ERROR;
-            hasResult = true;
+            return new JsonResult<>(JsonResult.ErrorCode.VALIDATION_ERROR, false);
         }
-        JsonResult<Boolean> result = new JsonResult<>(errorCode);
-        result.setResult(hasResult);
-        return result;
+        MultipartFile file = uploadFileDTO.getFile();
+        String uploadFileName = ObjectUtils.firstNonNull(uploadFileDTO.getLabel(), file.getOriginalFilename());
+
+        File tempFile = File.createTempFile(uploadFileName, "");
+        tempFile.deleteOnExit();
+        file.transferTo(tempFile);
+
+        if (archive) {
+            submissionService.uploadResultsByDataOwner(id, tempFile);
+            return new JsonResult<>(JsonResult.ErrorCode.NO_ERROR, true);
+        }
```

Comment:
In my opinion, a single point of exit typically makes code simpler to read and change. So what do you think of the following instead?
```
if (archive) {
    submissionService.uploadResultsByDataOwner(id, tempFile);
} else {
    submissionService.uploadResultsByDataOwner(id, uploadFileName, tempFile);
}
return new JsonResult<>(JsonResult.ErrorCode.NO_ERROR, true);
```

Reply:
Yeah, I agree here it is unnecessary duplication, but in general, return condition can really simplify complication if condition.